### PR TITLE
Add contributing docs

### LINKED ISSUE #1627: Add "Contribution guidelines" to the documentation
Inspiration: https://www.python-httpx.org/contributing/

If people are positive about https://github.com/encode/starlette/discussions/1623, it should be a mention there.

### PR DESCRIPTION
**Pull request by aminalaee**
Fixes #1627

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+- Code snippet
+- Error traceback
+
+You should always try to reduce any examples to the *simplest possible case*
+that demonstrates the issue.
+
+## Development
+
+To start developing Starlette create a **fork** of the
+[Starlette repository](https://github.com/encode/starlette) on GitHub.
```
Comment (Kludex) [inline docs/contributing.md]:
```suggestion
To start developing Starlette, create a **fork** of the
[Starlette repository](https://github.com/encode/starlette) on GitHub.
```

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+For an example, see [#1600](https://github.com/encode/starlette/pull/1600).
+
+Once the release PR is merged, create a
+[new release](https://github.com/encode/starlette/releases/new) including:
+
+- Tag version like `0.13.3`.
+- Release title `Version 0.13.3`
+- Description copied from the changelog.
+
+Once created this release will be automatically uploaded to PyPI.
```
Comment (Kludex) [inline docs/contributing.md:167]: There's just one more detail.

Not sure if good to mention, but to deploy to PyPI an [encode/operator](https://github.com/orgs/encode/teams/operations) needs to approve the deployment.

Comment (aminalaee) [reply docs/contributing.md:167]: Yeah, I think that would be more accurate. But I just kept it consistent with HTTPX, we can add it.

**Review by Kludex — APPROVED** [review]
Thanks @aminalaee 🙏

**Pull request merged**